torch/_dynamo/variables/tensor.py

```diff
@@ -1396,3 +1396,8 @@ def reconstruct(self, codegen):
         codegen(self.from_tensor)
         codegen.load_method("untyped_storage")
         codegen.call_method(0)
+
+
+class DataPtrVariable(VariableTracker):
+    def __init__(self, tensor: TensorVariable) -> None:
+        self.tensor = tensor
```

Comment:
This could also be in its own change, there are a couple of things that can go wrong here.

1) there should be a reconstruct method on this. The test case is:
```
@torch.compile(backend=eager)
def f(x):
   return x.data_ptr()
```

 and 

```
@torch.compile(backend=eager)
def f(x):
   y = x.data_ptr()
   torch._dynamo.graph_break()
   return x.clone()
```

2) if data_ptr is passed into a downstream operator (like torch.add?), Dynamo should handle this gracefully (by graph breaking). I'm not sure that's what happens today, can we add a test for it?